apps/api/src/domain/models/bibliographic-citation/journal-article-bibliographic-citation/entities/journal-article-bibliographic-citation.from-event-history.spec.ts

```diff
@@ -1,24 +1,234 @@
+
+                const journalArticle = result as JournalArticleBibliographicCitation;
+
+                expect(journalArticle.published).toBe(true);
+            });
+        });
+
+        describe(`when a user has been granted access`, () => {
+            it(`should return a journal article that allows access to the given user`, () => {
+                const result = JournalArticleBibliographicCitation.fromEventHistory(
+                    readAccessGranted.as(aggregateCompositeIdentifier),
+                    journalId
+                );
+
+                expect(result).toBeInstanceOf(JournalArticleBibliographicCitation);
+
+                const journalArticle = result as JournalArticleBibliographicCitation;
+
+                expect(journalArticle.queryAccessControlList.canUser(userId)).toBe(true);
+            });
         });
     });
 
     describe(`when the event history is invalid`, () => {
         describe(`when there are no events for the given Journal Article Bibliographic Citation`, () => {
-            it.todo(`should return Not Found`);
+            it(`should return Not Found`, () => {
+                const result = JournalArticleBibliographicCitation.fromEventHistory(
+                    journalArticleBibliographicCitationCreatedWithRequiredPropsOnly.as({
+                        type: AggregateType.bibliographicCitation,
+                        id: buildDummyUuid(980),
+                    }),
+                    journalId
+                );
+
+                expect(result).toBe(NotFound);
+            });
         });
 
         describe(`when the creation event is missing`, () => {
-            it.todo(`should throw`);
+            it(`should throw`, () => {
+                const eventSource = () => {
+                    JournalArticleBibliographicCitation.fromEventHistory(
+                        // remove the creation event
+                        readAccessGranted.as(aggregateCompositeIdentifier).slice(1),
+                        journalId
+                    );
+                };
+
+                expect(eventSource).toThrow();
+            });
         });
 
         // This could happen if we change our invariant rules or botch event versioning
         describe(`when there is a creation event with invalid data`, () => {
-            it.todo(`should return an invariant validation error`);
+            it(`should throw an invariant validation error`, () => {
+                const eventSource = () => {
+                    JournalArticleBibliographicCitation.fromEventHistory(
+                        new TestEventStream()
+                            .andThen<JournalArticleBibliographicCitationCreated>({
+                                type: 'JOURNAL_ARTICLE_BIBLIOGRAPHIC_CITATION_CREATED',
+                                payload: {
+                                    // this shouldn't be an empty string
+                                    title: '',
```

Comment:
Would it be worth creating a `invalidEmptyString` variable or something like that for this case?